Classes/TYPO3/TYPO3CR/Search/Indexer/AbstractNodeIndexer.php

```diff
@@ -86,10 +86,41 @@ protected function evaluateEelExpression($expression, Node $node, $propertyName,
 	 * @param Node $node
 	 * @param string $propertyName
 	 * @param string $fulltextExtractionExpression
+	 * @param array array<TYPO3\Media\Domain\Model\AssetInterface>
+	 * @throws IndexingException
+	 * @return array
+	 */
+	protected function extractAssetsToFulltext(Node $node, $propertyName, $fulltextExtractionExpression, array &$fulltextIndexOfNode) {
+
+		if ($fulltextExtractionExpression !== '') {
+
+			$extractedFulltext = $this->evaluateEelExpression($fulltextExtractionExpression, $node, $propertyName, ($node->hasProperty($propertyName) ? $node->getProperty($propertyName) : NULL));
+
+			if (!is_array($extractedFulltext)) {
+				throw new IndexingException('The fulltext index for property "' . $propertyName . '" of node "' . $node->getPath() . '" could not be retrieved; the Eel expression "' . $fulltextExtractionExpression . '" is no valid fulltext extraction expression.');
+			}
+
+			foreach ($extractedFulltext as $bucket => $value) {
+				if (!isset($fulltextIndexOfNode[$bucket])) {
+					$fulltextIndexOfNode[$bucket] = array();
+				}
+				$fulltextIndexOfNode[$bucket] = $value ;
+			}
+		}
+
+		return $fulltextIndexOfNode;
+	}
```

Comment:
First I tried that, to add them as a part of something like "assetParts". But this would mean that big parts must be rewritten. Or at least I didn't find a proper way without rewriting it.

One UseCase is also to index AssetLists with its filenames or whole TextWithImage Elements. So you would be able to bost the document if in the caption or title tag the search phrase is also to find.

Stuf like that

It is also really hard to describe all the things by writing, but I would love to have a discussion about that. In my opinion it is really a big feature with a lot dependencies.

Greets Markus